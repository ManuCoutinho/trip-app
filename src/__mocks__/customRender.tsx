import { render } from '@testing-library/react'
import { RenderOptions } from '@testing-library/react/types'
import { ThemeContextProvider } from 'contexts/ThemeContext'
import { ReactElement, FC, ReactNode } from 'react'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
