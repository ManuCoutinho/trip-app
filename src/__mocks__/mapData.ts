import { MapProps } from 'components/Map/types'

export default {
  places: [
    {
      id: 'first-place',
      name: 'Rio de Janeiro',
      slug: 'rio-de-janeiro',
      location: {
        latitude: -22.925453,
        longitude: -43.277763
      }
    },
    {
      id: 'second-place',
      name: 'Lagos',
      slug: 'lagos',
      location: {
        latitude: 6.460649,
        longitude: 3.333309
      }
    }
  ]
} as MapProps
