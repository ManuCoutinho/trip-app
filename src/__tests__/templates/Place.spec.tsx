import PlaceTemplate from 'templates/Place'
import { render } from '__mocks__/customRender'
import { renderHook, screen } from '@testing-library/react'
import placePropsMock, { placeWithOutHtml } from '__mocks__/data/placeMock'
import mockRouter from 'next-router-mock'
import { useRouter } from 'next/router'

jest.mock('next/router', () => require('next-router-mock'))

describe('<PlaceTemplate />', () => {
  it('should render the Place template correctly', () => {
    const { getByRole, getByTestId, getByText } = render(
      <PlaceTemplate {...placePropsMock} />
    )
    expect(getByTestId('back-link')).toBeInTheDocument()
    expect(getByRole('link', { name: /excursionista/i })).toBeInTheDocument()
    expect(getByText(/curtiu/i)).toBeInTheDocument()
  })
  it('should return null when isFallback is equal true', async () => {
    const { result } = renderHook(() => useRouter())
    console.log(result.current.isFallback)
    const { getByText } = render(<PlaceTemplate {...placeWithOutHtml} />)
    screen.debug()
  })
  it('should match to snapshot', () => {
    const { container } = render(<PlaceTemplate {...placePropsMock} />)
    expect(container).toMatchSnapshot()
  })
})
