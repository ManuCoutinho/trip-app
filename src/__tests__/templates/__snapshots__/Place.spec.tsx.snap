// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaceTemplate /> should match to snapshot 1`] = `
<div>
  <section
    class="sc-gswNZR hQQdUk"
  >
    <div
      class="sc-bcXHqe eCYpHH"
      data-testid="back-link"
    >
      <a
        href="/"
        title="link to Home"
      >
        <svg
          aria-label="Home"
          fill="currentColor"
          height="32"
          viewBox="0 0 256 256"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <line
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
            x1="200"
            x2="56"
            y1="56"
            y2="200"
          />
          <line
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
            x1="200"
            x2="56"
            y1="200"
            y2="56"
          />
        </svg>
      </a>
    </div>
    <div
      class="sc-dkrFOg cJzsCG"
    >
      <h1
        class="sc-eDvSVe kAHOWE"
      >
        Rio de Janeiro
      </h1>
      <div
        class="sc-jSUZER gMLWgb"
      >
        some description about the city here
      </div>
      <div
        class="sc-gKPRtg fVssuT"
      >
        <img
          alt="Rio de Janeiro"
          data-nimg="1"
          decoding="async"
          height="600"
          loading="lazy"
          src="/_next/image?url=%2Fpic-rj.jpg&w=2048&q=75"
          srcset="/_next/image?url=%2Fpic-rj.jpg&w=1080&q=75 1x, /_next/image?url=%2Fpic-rj.jpg&w=2048&q=75 2x"
          style="color: transparent;"
          width="1000"
        />
      </div>
    </div>
    <a
      href="/undefined/gallery?query=Rio%20de%20Janeiro"
      rel="noopener noreferrer"
      target="_blank"
      title="Excursionista"
    >
      <p
        class="sc-hLBbgP jtzTbC"
      >
        Curtiu ? Ent√£o veja mais destinos...
      </p>
    </a>
    <span
      class="sc-iBYQkv flWBRS"
    >
      <a
        href="/#"
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <circle
            cx="128"
            cy="128"
            fill="none"
            r="96"
            stroke="currentColor"
            stroke-miterlimit="10"
            stroke-width="16"
          />
          <polyline
            fill="none"
            points="94.1 121.9 128 88 161.9 121.9"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <line
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
            x1="128"
            x2="128"
            y1="168"
            y2="88"
          />
        </svg>
      </a>
    </span>
  </section>
</div>
`;
